core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -251,7 +254,9 @@ private RexNode simplifyComparison(RexCall e) {
     // "1 != '1'" is unchanged because the types are not the same.
     if (o0.isA(SqlKind.LITERAL)
         && o1.isA(SqlKind.LITERAL)
-        && o0.getType().equals(o1.getType())) {
+        // compare type exclude nullability here
```

Comment:
This comment clarifies nothing.
Please either remove it, or clarify WHY nullability is not important here.